spec/src/consensus.rs

```diff
@@ -778,6 +779,14 @@ impl Consensus {
             self.primary_epoch_reward(epoch.number() + 1)
         }
     }
+
+    pub fn is_mainnet(&self) -> bool {
+        *IS_MAINNET.get_or_init(|| {
+            let mainnet_genesis_hash: H256 =
+                h256!("0x92b197aa1fba0f63633922c61c92375c9c074a93e85963554f5499fe1450d0e5");
```

Comment:
suggest to use bundled resouce file and compare genesis hash:
```
spec = ChainSpec::load_from(Resource::bundled("specs/mainnet.toml"))
self. genesis_hash  == spec.genesis
```

Reply:
we may add a `mainnet` bool field to `Consensus` struct and move this comparison logic code to `ConsensusBuilder#build`, it will remove the `once_cell` deps.

Reply:
I recommend adding it to chain spec as trivial options. I define trivial options as the ones does not affect consensus, thus they can be ignored when computing hash in #2255